.github/workflows/cypress-test.yml

```diff
@@ -16,22 +16,45 @@ env:
 jobs:
   tests:
     name: Run Cypress tests
-    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest , windows-latest ]
+        jdk: [ 11, 17 ]
+    runs-on: ${{ matrix.os }}
+
     steps:
+      - name: Set up JDK
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.jdk }}
+
       - name: Checkout Branch
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
       
-      - name: Download OpenSearch Security Plugin
+      - name: Download OpenSearch Security Plugin on Linux
+        if: ${{ runner.os == 'Linux' }}
         run: wget --progress=bar:force:noscroll -O opensearch-security.zip https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/${{ env.OPENSEARCH_VERSION }}/latest/linux/x64/tar/builds/opensearch/plugins/${{ env.PLUGIN_NAME }}-${{ env.PLUGIN_VERSION }}.zip
       
-      - name: Create Setup Script
+      - name: Download OpenSearch Security Plugin on Windows
+        if: ${{ runner.os == 'Windows' }}
+        run: Invoke-WebRequest -Uri https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/${{ env.OPENSEARCH_VERSION }}/latest/windows/x64/zip/builds/opensearch/plugins/${{ env.PLUGIN_NAME }}-${{ env.PLUGIN_VERSION }}.zip  -Outfile opensearch-security.zip
```

Comment:
Use `peternied/download-file` [(link)[https://github.com/peternied/download-file]] so the command you execute is the same.  Use `action-cond` to set the different platform urls to compact this down

Reply:
Done.